lib-impl/src/main/java/org/ergoplatform/appkit/impl/Eip29AttachmentBuilder.java

```diff
@@ -0,0 +1,65 @@
+package org.ergoplatform.appkit.impl;
+
+import static org.ergoplatform.appkit.impl.Eip4TokenBuilder.getSerializedErgoValueForRegister;
+
+import org.ergoplatform.ErgoBox;
+import org.ergoplatform.appkit.Eip29Attachment;
+import org.ergoplatform.appkit.ErgoValue;
+import org.ergoplatform.appkit.GenericEip29Attachment;
+import org.ergoplatform.appkit.TransactionBox;
+import org.ergoplatform.explorer.client.model.AdditionalRegisters;
+
+import java.util.List;
+
+import javax.annotation.Nonnull;
+import javax.annotation.Nullable;
+
+public class Eip29AttachmentBuilder {
+
+    /**
+     * @return EIP-29 compliant attachment built from serialized ergo value
+     */
+    public static Eip29Attachment buildFromHexEncodedErgoValue(String hexEncodedErgoValue) {
+        return GenericEip29Attachment.createFromErgoValue(ErgoValue.fromHex(hexEncodedErgoValue));
+    }
+
+    /**
+     * @return EIP-29 compliant attachment, if found. null otherwise
+     */
+    @Nullable
+    public static Eip29Attachment buildFromAdditionalRegisters(@Nonnull AdditionalRegisters registers) {
+        try {
+            return buildFromHexEncodedErgoValue(getSerializedErgoValueForRegister(registers, "R9"));
```

Comment:
Thinking more about this R9 hard coded constant I think it is better to avoid such hardcoding entirely. You can obtain the name of the last register using methods in ErgoBox. 
In addition to make sure the implementation satisfies the EIP, a test can be added which checks that the method getLastRegisterName == "R9"

Reply:
Added this, but I am not sure if it is a change for the better. At the end of the day, it is still a constant but just more obfuscated.

Reply:
I would say it is a constant expression which depends on other constants. Theoretically, in future, we may increase number of registers (highly unlikely) and we want attachment to alway be the last register.